interface/modules/custom_modules/oe-module-payroll/src/Bootstrap.php

```diff
@@ -0,0 +1,269 @@
+
+        foreach ($settings as $key => $config) {
+            $value = $GLOBALS[$key] ?? $config['default'];
+            $service->appendToSection(
+                $section,
+                $key,
+                new GlobalSetting(
+                    xlt($config['title']),
+                    $config['type'],
+                    $value,
+                    xlt($config['description']),
+                    true
+                )
+            );
+        }
+    }
+
+
+    /**
+     * @param TwigEnvironmentEvent $event
+     */
+    public function addTemplateOverrideLoader(TwigEnvironmentEvent $event)
+    {
+        try {
+            $twig = $event->getTwigEnvironment();
+            if ($twig === $this->twig) {
+                // we do nothing if its our own twig environment instantiated that we already setup
+                return;
+            }
+            // we make sure we can override our file system directory here.
+            $loader = $twig->getLoader();
+            if ($loader instanceof FilesystemLoader) {
+                $loader->prependPath($this->getTemplatePath());
+            }
+        } catch (LoaderError $error) {
+            $this->logger->errorLogCaller("Failed to create template loader", ['innerMessage' => $error->getMessage(), 'trace' => $error->getTraceAsString()]);
+        }
+    }
+
+    public function registerMenuItems(): void
+    {
+            /**
+             * @var EventDispatcherInterface $eventDispatcher
+             * @var array $module
+             * @global                       $eventDispatcher @see ModulesApplication::loadCustomModule
+             * @global                       $module @see ModulesApplication::loadCustomModule
+             */
+            $this->eventDispatcher->addListener(MenuEvent::MENU_UPDATE, [$this, 'addProviderRateMenuItem']);
+        $this->eventDispatcher->addListener(MenuEvent::MENU_UPDATE, [$this, 'adminEarningsReportMenuItem']);
+        $this->eventDispatcher->addListener(MenuEvent::MENU_UPDATE, [$this, 'providerViewEarningsReportMenuItem']);
+    }
+
+    public function addProviderRateMenuItem(MenuEvent $event): MenuEvent
+    {
+        $menu = $event->getMenu();
+
+        $menuItem = new \stdClass();
+        $menuItem->requirement = 0;
+        $menuItem->target = 'adm';
+        $menuItem->menu_id = 'adm0';
+        $menuItem->label = xlt("Set Providers Rates");
+        // TODO: pull the install location into a constant into the codebase so if OpenEMR changes this location it
+        // doesn't break any modules.
+        $menuItem->url = "/interface/modules/custom_modules/oe-module-payroll/public/provider_rate/index.php";
+        $menuItem->children = [];
+
+        /**
+         * This defines the Access Control List properties that are required to use this module.
+         * Several examples are provided
+         */
+        $menuItem->acl_req = ["admin", "super"];
```

Comment:
for this script and other scripts ensure you also do an access control on the script itself (ie. make sure provider_rate/index.php does not work if does not have admin, super acl)